resources/metadata/artifactPrepareVersion.yaml

```diff
@@ -209,6 +210,25 @@ spec:
           - PARAMETERS
           - STAGES
           - STEPS
+      - name: CAPVersioningPreference
+        type: string
+        description: "For CAP build tool only: Defines which file should be used for versioning."
+        longDescription: |
+          If `maven` is chosen (default value), then the step expects a pom.xml file in the project's root folder. Alternatively, you can specify the path to the file using the `filePath` parameter.
+          If `npm` is chosen, then the step expects a package.json file in the project's root folder. Alternatively, you can specify the path to the file using the `filePath` parameter.
+
+          In case you want to propagate version to addition file(s), it can be done via `additionalTargetTools` and `additionalTargetDescriptors` parameters.
```

Comment:
An additional note mentioning CAP projects which use just npm modules can not use this param would help? or am I wrong?

Reply:
even for such projects those params can be used if needed. For example to propagate version to docker build descriptor file (VERSION) or helm (Chart.yaml)